src/Misc/screenshotTools.ts

```diff
@@ -126,12 +126,18 @@ export class ScreenshotTools {
 
         var scene = camera.getScene();
         var previousCamera: Nullable<Camera> = null;
+        let previousActiveCameras: Camera[] = [];
 
         if (scene.activeCamera !== camera) {
             previousCamera = scene.activeCamera;
             scene.activeCamera = camera;
         }

```

Comment:
Shouldn't this already handle multi-camera setup?

Reply:
Nope, in fact, it is only needed to work with multi-camera setup. That is because `activeCameras` has I higher priority for both. 

This is because of this code:
https://github.com/BabylonJS/Babylon.js/blob/130f91dd9eb5becb2495737013aee5b46111b665/src/scene.ts#L3986

I am testing and sending a refactored code that simplifies the solution

Reply:
Shouldn't the call of `scene.render()` necessary? Because:

- if Yes, we should only work with `activeCameras`;
- if No, we should remove the `scene.render()` call.

Reply:
works good, of course the most performant method would be to remove the call.
but better wait for core members, shouldn't be more than a couple hours max, 
maybe there's a good reason it's there and has to stay

Reply:
In fact, my simplified solution doesn't work calling an hidden camera in single camera scene. My fancy solution solves that: https://playground.babylonjs.com/#Y0CG29#3 
(space bar will capture screenshot)

So, we have two ways. Removing the call of `scene.render()` or cleaning `activeCameras` also.

Reply:
as a note, you can actually just move the __scene.activeCamera__ check below __scene.render()__ aswell (in-case it must stay)
https://playground.babylonjs.com/#Y0CG29#4

Reply:
I kind of like this fix maybe a bit better as it does not create a new temp array

Reply:
@deltakosh so, with your answer:
> yes @gabrielheming yes this is mandatory to update everything (matrices, viewport, cache, etc..)

I have to use the temp array, because the render is called after changing the camera.

Reply:
I mean with @aWeirdo  fix we do not need to store the previousActiveCameras at all

Reply:
That's why I asked if it was mandatory call `scene.render()` **after** changing the activeCamera. That solution call `scene.render()` **before**, and does not work calling after.

Reply:
Oh ok..yeah! calling before is totally fine

Reply:
Good. I will update it.